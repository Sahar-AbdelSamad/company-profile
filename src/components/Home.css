.homeSection1 {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  height: 100px;
}

.homeSection1 ul img {
  height: 50px;
  width: 50px;
}

.homeSection1 li {
  font-weight: 700;
  font-size: larger;
  list-style-type: none;
}

.search-bar input,
.search-bar select {
  border: none;
  padding-left: 1%;
  height: 20px;
}

.search-bar input {
  width: 60%;
  color: #1f4180;
}

.search-bar select {
  width: 40%;
  background-color: #1f4180;
  color: white;
}

.cards {
  display: flex;
  flex-wrap: wrap;
}

.cards a {
  width: 50%;
  text-decoration: none;
  height: 150px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding-left: 10px;
}

.cards li {
  color: white;
  list-style-type: none;
  font-weight: 300;
}

.cards .companyName {
  font-weight: 700;
}

.cards a:nth-child(4n+1),
.cards a:nth-child(4n+4) {
  background-color: #1f4180;
}
