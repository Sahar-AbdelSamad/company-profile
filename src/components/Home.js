import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { fetchData } from '../redux/company/company';
import './Home.css';

const Home = () => {
  const home = useSelector((state) => state.homeReducer);
  const [selected, setSelected] = useState('Choose a company');
  const dispatch = useDispatch();
  const select = (e) => {
    const str = e.target.value;
    setSelected(str);
    const mySubString = str.substring(
      str.indexOf('(') + 1,
      str.lastIndexOf(')'),
    );
    dispatch(fetchData(mySubString));
  };
  return (
    <div>
      <section className="home">Section1</section>
      <section className="search-bar">
        <select id="company" name="company" value={selected} onChange={select}>
          <option value="Choose a company" disabled>Choose a company</option>
          {Object.entries(home).map(([key, value]) => (
            <option key={key} value={`${value.name} (${value.symbol})`}>
              {`${value.name} (${value.symbol})`}
            </option>
          ))}
        </select>
      </section>
      <NavLink to="/Company">{selected}</NavLink>
      <section className="cards">
        {Object.entries(home).map(([key, value]) => (
          <NavLink key={key} to="/Company">
            <div
              role="button"
              tabIndex={0}
              onKeyDown={() => dispatch(fetchData(value.symbol))}
              onClick={() => dispatch(fetchData(value.symbol))}
            >
              {value.name}
              <br />
              {value.symbol}
              <br />
              {value.exchange}
              <br />
            </div>
          </NavLink>
        ))}
      </section>
    </div>
  );
};

export default Home;
