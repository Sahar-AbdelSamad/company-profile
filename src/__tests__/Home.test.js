import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from '../components/Home';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
