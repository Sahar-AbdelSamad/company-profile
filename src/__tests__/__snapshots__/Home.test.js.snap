// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="header"
  >
    <p />
    <p>
      Companies
    </p>
    <ul
      className="icons"
    >
      <li>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
          />
        </svg>
      </li>
      <li>
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </li>
    </ul>
  </header>,
  <section
    className="homeSection1"
  >
    <ul
      className="nasdaq"
    >
      <li>
        Nasdaq-100
      </li>
      <li>
        <img
          alt="Nasdaq-100"
          src="nasdaq.png"
        />
      </li>
    </ul>
    <ul
      className="dowjones"
    >
      <li>
        Dow Jones Industrial Average
      </li>
      <li>
        <img
          alt="Dow Jones Industrial Average (DJIA)"
          src="dowjones.png"
        />
      </li>
    </ul>
  </section>,
  <section
    className="search-bar"
  >
    <input
      onChange={[Function]}
      placeholder="Search Company Name or Symbol"
      type="search"
      value=""
    />
    <select
      id="company"
      name="company"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        Choose a company
      </option>
    </select>
  </section>,
  <section
    className="cards"
  />,
]
`;
